api/server/router/volume/volume_routes.go

```diff
@@ -517,6 +523,10 @@ func (r *router) volumeAttach(
 			}
 		}
 
+		if v.AttachmentState == 0 {
+			v.AttachmentState = types.VolumeAttached
+		}
+
```

Comment:
Although now that you point this out, I need to update the above to the following logic:

```
if v.AttachmentState == 0 && len(v.Attachments) > 0 {
    if "attached to the same instance as the instance ID in the context" {
        v.AttachmentState = types.VolumeAttached
    } else {
        v.AttachmentState = types.VolumeUnavailable
    }
}